### ISSUE TITLE
recon/domains-hosts/netcraft - Not working

### ISSUE THREAD

**Issue by MrJester** [lanmaster53/recon-ng-marketplace#30]
```
[recon-ng][default] > marketplace install recon/domains-hosts/netcraft
[*] Module installed: recon/domains-hosts/netcraft
[*] Reloading modules...
[recon-ng][default] > modules load recon/domains-hosts/netcraft
[recon-ng][default][netcraft] > info

      Name: Netcraft Hostname Enumerator
    Author: thrapt (thrapt@gmail.com)
   Version: 1.0

Description:
  Harvests hosts from Netcraft.com. Updates the 'hosts' table with the results.

Options:
  Name    Current Value  Required  Description
  ------  -------------  --------  -----------
  SOURCE  default        yes       source of input (see 'show info' for details)

Source Options:
  default        SELECT DISTINCT domain FROM domains WHERE domain IS NOT NULL
  <string>       string representing a single input
  <path>         path to a file containing a list of inputs
  query <sql>    database query returning one column of inputs

[recon-ng][default][netcraft] > run
[!] 'Response' object has no attribute 'cookiejar'.
[!] Something broken? See https://github.com/lanmaster53/recon-ng/wiki/Troubleshooting#issue-reporting.
```

**Comment by azRKI3H72go4QUPe** [comment]
Any updates on this? Still also getting the 'cookiejar' error.

**Comment by MrJester** [comment]
There has not been but we would love any help if you wanted to submit a pull request.

**Comment by lanmaster53** [comment]
Let's try to get @thrapt's attention for this one.

**Comment by safesecs** [comment]
Just for visibility, happening here as well.  Its ok, I finally figured out t he version differences... duh.  This may actually be valid!  :)

[recon-ng][Testing][netcraft] > run
[!] 'Response' object has no attribute 'cookiejar'.
[!] Something broken? See https://github.com/lanmaster53/recon-ng/wiki/Troubleshooting#issue-reporting.

**Comment by safesecs** [comment]
Wonder if the "response" error would be the challenge cookie in the .py....

for cookie in cookiejar:
            if cookie.name == 'netcraft_js_verification_challenge':
                challenge = cookie.value
                response = hashlib.sha1(unquote_plus(challenge)).hexdigest()
                cookiejar.set_cookie(self.make_cookie('netcraft_js_verification_response', f"{response}", '.netcraft.com'))
                break

**Comment by kakashimunoz20** [comment]
I got the same error, any updates?

**Comment by lanmaster53** [comment]
Let's ping @thrapt again. If he isn't responsive, perhaps someone else would like to take a shot at fixing it?

**Comment by rezavol** [comment]
@thrapt
i'm gonna have to link this in my homework for explaning it's not working, bad luck

**Comment by lanmaster53** [comment]
It's a pretty bold move to use something like this as part of a long term lesson plan. These resources are constantly changing. I would advise your teacher/professor to be more general about the assignment and emphasize the data over the module used to harvest it.

**Comment by mavensecurity** [comment]
It seems netcraft only loves real browsers, the kind that can solve JS math puzzles to prove they are not bots/scripts.  Sounds like a job for Headless Chromium (phantomJS is no more).  Here's a recon-ng dump from setting VERBOSITY to 2.  I also changed my USER-AGENT just in case they were filtering on that.  I don't know why this looks like a query for "test.com" because that was not the domain I had set as SOURCE.

[recon-ng][Power1][netcraft] > run
[*] DATABASE => /home/david/.recon-ng/workspaces/Power1/data.db
[*] QUERY => INSERT OR REPLACE INTO dashboard (module, runs) VALUES ('recon/domains-hosts/netcraft', COALESCE((SELECT runs FROM dashboard WHERE module='recon/domains-hosts/netcraft')+1, 1))
[*] ========================= REQUEST =========================
url:    https://searchdns.netcraft.com/?restriction=site%252Bends%252Bwith&host=test.com
method: GET /?restriction=site%252Bends%252Bwith&host=test.com
header: User-Agent: "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:70.0) Gecko/20100101 Firefox/70.0"
header: Accept-Encoding: gzip, deflate
header: Accept: */*
header: Connection: keep-alive
[*] ========================= RESPONSE =========================
status: 200 OK
header: Date: Fri, 08 Nov 2019 02:01:57 GMT
header: Server: Apache
header: Set-Cookie: netcraft_js_verification_challenge=djF8K0tZU296RW1JWE4wNXExSDJvakJtSm1YYU5MMTd4aFBCSUdHKzEvVmord29tMm5lcjA1MEFj%0AT0JZdzRyYjRQUjh0RTdURkp5bS9KQgpoVDVHdFVDK213PT0KfDE1NzMxNzg1MTc%3D%0A%7C564284851608559752188acfe063853ea307308a; path=/
header: X-Frame-Options: SAMEORIGIN
header: X-XSS-Protection: 1; mode=block
header: X-Content-Type-Options: nosniff
header: Accept-Ranges: bytes
header: Content-Length: 2123
header: Cache-Control: max-age=0
header: Expires: Fri, 08 Nov 2019 02:01:57 GMT
header: Connection: close
header: Content-Type: text/html; charset=UTF-8
body:   b'<!DOCTYPE html>\n<html>\n    <head>\n    </head>\n    <body>\n        <script\n            src="/errors/sha1.js"></script>\n        <script>\n\n            if (self == top) {\n                set_verification_cookie();\n            }\n            else {\n                top.location = "about:blank";\n            }\n\n\n            function set_verification_cookie() {\n                var challenge_token_present = false;\n\n                var cookies = document.cookie.split(/\\s*;\\s*/);\n                for(var i = 0; i < cookies.length; i++) {\n                    var cookie = cookies[i].split(\'=\');\n\n                    if(cookie[0] == "netcraft_js_verification_challenge") {\n                        var challenge_token = unescape(cookie[1]);\n                        var response = CryptoJS.SHA1(challenge_token);\n\t\t\t            challenge_token_present = true;\n                    }\n                }\n\n                if ( challenge_token_present ) {\n                    document.cookie = "netcraft_js_verification_response=" + response + "; path=/";\n                    if ( \'/errors/challenge.html\' != window.location.pathname ) {\n                        document.location.reload();\n                    }\n                }\n                else {\n                    document.write("<h1>Your browser does not support Cookies</h1>");\n                    document.write("<p>Certain pages on netcraft.com require the use of Cookies to distinguish between robotic users and genuine visitors.</p>");\n                    document.write("<p>Please re-enable Cookies (it\'s fine to reject 3rd party cookies) then try again</p>");\n                }\n            }\n        </script>\n\n        <noscript id="explain">\n        <h1>Your browser does not support JavaScript and/or Cookies. </h1>\n        <p>Certain pages on\n            *.netcraft.com require the use of both JavaScript and Cookies to distinguish\n            between robotic users and genuine visitors.\n            </p>\n            <p>Please re-enable JavaScript and Cookies (it\'s fine to\n            reject 3rd party cookies) then try again</p>\n        </noscript>\n\n    </body>\n</html>\n'
------------------------------------------------------------
Traceback (most recent call last):
  File "/home/david/Tools/recon-ng/recon/core/module.py", line 299, in do_run
    self.run()
  File "/home/david/Tools/recon-ng/recon/core/module.py", line 293, in run
    self.module_run(*params)
  File "/home/david/.recon-ng/modules/recon/domains-hosts/netcraft.py", line 26, in module_run
    cookiejar = resp.cookiejar
AttributeError: 'Response' object has no attribute 'cookiejar'
------------------------------------------------------------
[!] Something broken? See https://github.com/lanmaster53/recon-ng/wiki/Troubleshooting#issue-reporting.

**Comment by jfbaillette** [comment]
```
root@kali:~# apt-get install python-cookies
```

**Comment by AissaouiD** [comment]
Any updates on this? Still also getting the 'cookiejar' error.

**Comment by lanmaster53** [comment]
Still needs to be fixed. Modules are by the community for the community. Feel free to give it a shot.

**Comment by shadyboy313** [comment]
Any updates?

**Comment by MrJester** [comment]
The only update is that the issue isn’t closed because no one in the community has been able to submit a pull request nor have any of us had time to review this. Would love if you could give it a shot at helping out.

**Comment by MrJester** [comment]
@cam-barts While I appreciate you linking to your pull request that is not a proper fix for this module. While it fixes the cookiejar error it does not fix the module. The module parsing needs to be updated to match the changes that netcraft has implemented.

**Comment by leebaird** [comment]
9.5 months later and this module is still broken. Suggest fixing or removing.

**Comment by lanmaster53** [comment]
@leebaird It took less characters to fix it than it took to write your response. You could have been the one to fix it 9.5 months ago! ;-)
